pyte/screens.py

```diff
@@ -359,8 +359,9 @@ def set_mode(self, *modes, **kwargs):
 
         # Mark all displayed characters as reverse.
         if mo.DECSCNM in modes:
-            self.buffer[:] = ([char._replace(reverse=True) for char in line]
-                              for line in self.buffer)
+            for line in self.buffer:
+                for char in self.buffer[line]:
+                    self.buffer[line][char]._replace(reverse=True)
```

Comment:
This does not alter the character, see docs for [`namedtuple._replace`](https://docs.python.org/3/library/collections.html#collections.somenamedtuple._replace).